### ISSUE TITLE
[1.6.2-beta.12] On mobile it's always loading the legacy package

### ISSUE THREAD

**Issue by vuhrmeister** [meteor/meteor#9751]
I was curious and updated to 1.6.2-beta.12 to test it out. As expected on latest Chrome it loads the smaller bundle without that legacy stuff.

But on mobile it loads the legacy bundle. I'm on Android 8 with latest Chrome.
Also on Chrome for Mac when switching to mobile view it always loads the legacy bundle.

Is this intended? Did I miss to configure something? 
Especially on mobile I would care to load the smaller bundle.

**Comment by benjamn** [comment]
Yes, this is currently working as designed. Cordova bundles JS at build time, so we don't know yet whether the bundle is going to be requested by a modern client or a legacy client, so we have to assume the worst.

This is different from how it works in a desktop web browsers, where each client requests the appropriate JS bundle every time the page loads. In some sense, the Cordova application is serving as a kind of local web server for the JS bundle.

For what it's worth, since the JS bundle is included in the deployed application (rather than downloaded from a remote server), which is installed on the device, users don't have to wait longer to download a larger bundle. It just takes up more space on the device, and may take longer to parse/evaluate.

We could include both the modern and the legacy bundle in the Cordova application archive, but that would take up even more space on the device. However, If you're more concerned about parsing time or runtime performance of the JS (for modern mobile clients, at least), then this might be a reasonable tradeoff to consider.

Also note: even though we're referring to the `web.browser.legacy` and `web.cordova` bundles as "legacy," they are essentially equivalent to what was shipped to all clients in previous versions of Meteor (including 1.6.1), so they're not worse/larger than what you saw before (though I totally understand wishing they were smaller).

**Comment by vuhrmeister** [comment]
Thx for the detailed explanation. Though I'm not talking about Cordova. I don't even have a Cordova build. I do see that it's not crucial for Cordova since it's shipped with the app.

But is it the same thing for the normal Chrome app on mobile or even on desktop with mobile view activated? There it's loading the bundle also on every request.

**Comment by benjamn** [comment]
Ah, in that case I would try [running Meteor from a checkout](https://github.com/meteor/meteor/blob/devel/DEVELOPMENT.md#running-from-a-git-checkout) (from the `release-1.6.2` branch) and putting a `console.log` or breakpoint in the `isModern` function [here](https://github.com/meteor/meteor/blob/3ee4b61b1dde15e6f8129821caf697e62df4ffee/packages/modern-browsers/modern.js#L11) too see what `browser.{name,major,minor,patch}` comes through when you load the app in a mobile Chrome view. It's possible that the user agent is getting disqualified unnecessarily. Alternatively, you can just change that function to return `true` unconditionally to force using the modern bundle for all clients, to see if it works.

**Comment by vuhrmeister** [comment]
I debugged it the way you suggested. That's the outcome:

When using desktop Chrome, this is the `browser` information: 
```javascript
{ name: 'chrome', major: 65, minor: 0, patch: 3325 }
```
And this using mobile view: 
```javascript
{ name: 'chromeMobile', major: 65, minor: 0, patch: 3325 }
```

That's what `minimumVersions`  is:
```javascript
{
  chrome: {
    version: 55,
    source: '/node_modules/meteor/babel-compiler/versions.js'
  },
  edge: {
    version: 15,
    source: '/node_modules/meteor/babel-compiler/versions.js'
  },
  firefox: {
    version: 52,
    source: '/node_modules/meteor/babel-compiler/versions.js'
  },
  mobile_safari: {
    version: [10, 3],
    source: '/node_modules/meteor/babel-compiler/versions.js'
  },
  opera: {
    version: 42,
    source: '/node_modules/meteor/babel-compiler/versions.js'
  },
  safari: {
    version: [10, 1],
    source: '/node_modules/meteor/babel-compiler/versions.js'
  },
  phantomjs: {
    version: Infinity,
    source: '/node_modules/meteor/modern-browsers/modern.js (generator functions)'
  },
  node: {
    version: 8,
    source: '/node_modules/meteor/babel-compiler/versions.js'
  },
  ie: {
    version: 10,
    source: '/node_modules/meteor/socket-stream-client/server.js'
  }
}
```

So I would think `chromeMobile` should be added to that list.

I tested it with the modern package on my mobile phone and it actually works.

**Comment by benjamn** [comment]
Yeah, that makes sense!

Adding support for `chromeMobile` will require updating all the calls to `setMinimumBrowserVersions` across the codebase, mostly in [`modern-browsers/modern.js`](https://github.com/meteor/meteor/blob/release-1.6.2/packages/modern-browsers/modern.js) and [`babel-compiler/versions.js`](https://github.com/meteor/meteor/blob/release-1.6.2/packages/babel-compiler/versions.js), to enforce minimum `chromeMobile` versions for each language feature.

I'm guessing the minimum major version will end up being 55, since that's what MDN [shows](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) for `async` function support for Chrome on Android, but as long as it's less than or equal to 65, your device will be able to use the modern bundle.

**Comment by vuhrmeister** [comment]
Okay cool, will do that. I already seen all those places where the version needs to be added.

I would say we can align the `chromeMobile` versions with the desktop versions since the mobile version is based on the desktop version. I couldn't find though from which version on this is valid.
> https://developer.chrome.com/multidevice/faq
> How similar is Chrome for Android with the desktop/laptop version of Chrome?
> Chrome for Android is based on the same source as Chrome for desktop.

**Comment by benjamn** [comment]
Interesting… if we could just normalize `chromeMobile` to `chrome` and re-use those versions, that would be convenient!

**Comment by vuhrmeister** [comment]
Thx, that was a good input. Have a look at https://github.com/meteor/meteor/pull/9793. What do you think?

**Comment by vuhrmeister** [comment]
I just tested it on Chrome Mobile version 50. It loads the legacy package as expected 👍
